pennylane/_device.py

```diff
@@ -98,6 +98,15 @@ def _process_shot_sequence(shot_list):
     return total_shots, shot_vector
 
 
+def _get_num_copies(shot_vector):
+    """Helper function to determine the number of copies from a shot vector Sequence(int) or Sequence(ShotTuple)."""
```

Comment:
Is there a difference between this function and `len(device._raw_shot_sequence)`?

Reply:
Potentially _raw_shot can be potentially (100, (100, 5), 100) and therefore it would not work in this case.